src/components/ComponentSettingsModal.jsx

```diff
@@ -0,0 +1,467 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+import { Base64 } from 'js-base64';
+import PropTypes from 'prop-types';
+import * as _ from 'lodash';
+import { Redirect } from 'react-router-dom';
+import FormField from './FormField';
+import FormFieldFile from './FormFieldFile';
+import LoadingButton from './LoadingButton';
+import {
+  frontMatterParser,
+  concatFrontMatterMdBody,
+  enquoteString,
+  dequoteString,
+  generateResourceFileName,
+  generateCollectionPageFileName,
+  generatePageFileName,
+  retrieveCollectionAndLinkFromPermalink,
+} from '../utils';
+import elementStyles from '../styles/isomer-cms/Elements.module.scss';
+import { validatePageSettings, validateResourceSettings } from '../utils/validators';
+import DeleteWarningModal from './DeleteWarningModal';
+
+// axios settings
+axios.defaults.withCredentials = true
+
+export default class ComponentSettingsModal extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      permalink: '',
+      fileUrl: null,
```

Comment:
The following steps result in us losing the original state of the file URL:
1. Go to resources and click on the settings button for a downloadable file
2. Switch the post type to `Post Content`
3. Switch it back to `Downloadable File`

The original downloadable file is now gone

Reply:
Right, this is an issue for the original permalink as well. Resolved in https://github.com/isomerpages/isomercms-frontend/pull/184/commits/29c3f6b7033020c10ef856335de7d1d8f62e11f7

Reply:
Was only working intermittently previously. As discussed, changed to use onChange instead of onClick, resolved in https://github.com/isomerpages/isomercms-frontend/pull/184/commits/e81877aca06af7ffaf295425a020f36051fc0461